shared.js

```diff
@@ -217,6 +228,40 @@ class MemFS {
     return ESUCCESS;
   }
 
+  host_read(fd, iovs, iovs_len, nread) {
+    this.hostMem_.check();
+    assert(fd === 0);
+    let size = 0;
+    for (let i = 0; i < iovs_len; ++i) {
+      const data = this.hostMem_.read32(iovs);
+      iovs += 4;
+      const len = this.hostMem_.read32(iovs);
+      iovs += 4;
+      const lenToWrite = len < (this.stdinStr.length - this.stdinStrPos) ? len : (this.stdinStr.length - this.stdinStrPos);
+      if(lenToWrite === 0){
+        break;
+      }
+      const buf = this.hostMem_.read(data,len)
```

Comment:
Can we replace the code on lines 244 to 251 with this instead?

```js
this.hostMem_.write(data, this.stdinStr.substr(0, lenToWrite));
```

It seems like it should be equivalent, but I'm not sure.

Reply:
My fault, I thought [this line](https://github.com/binji/llvm-project/blob/master/binji/memfs.c#L645) was to copy the data of iovs from the host and I was copying it.

Reply:
And should it be like below?
```
this.hostMem_.write(data, this.stdinStr.substr(this.stdinStrPos, lenToWrite));
```
It's my original code's mistake. The offset weren't taken into consideration,

Reply:
Yes, I think you're right.